rclpy/rclpy/qos.py

```diff
@@ -308,3 +308,65 @@ def __init__(self, qos_profile: QoSProfile, profile_name: str):
     'qos_profile_parameter_events')
 qos_profile_action_status_default = _rclpy_action.rclpy_action_get_rmw_qos_profile(
     'rcl_action_qos_profile_status_default')
+
+
+class QoSNameTranslations:
```

Comment:
I would rather see these names become part of the enum class. I think if the name is closer to the enum it will be more likely to be updated if/when new QoS features are added. Also, personally I want shorter names in code.

```python3
#!/usr/bin/env python3

from enum import IntEnum


class QoSLivelinessPolicy(IntEnum):
    """
    Enum for QoS Liveliness settings.
    This enum matches the one defined in rmw/types.h
    """

    RMW_QOS_POLICY_LIVELINESS_SYSTEM_DEFAULT = 0
    system_default = RMW_QOS_POLICY_LIVELINESS_SYSTEM_DEFAULT 
    RMW_QOS_POLICY_LIVELINESS_AUTOMATIC = 1
    automatic = RMW_QOS_POLICY_LIVELINESS_AUTOMATIC 
    RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_NODE = 2
    manual_by_node = RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_NODE 
    RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC = 3
    manual_by_topic = RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC 


import code
code.interact(local=locals())

```

Enum classes are generated with a `__getitem__` that allows getting the member with a string.

```python3
>>> QoSLivelinessPolicy['system_default']
<QoSLivelinessPolicy.RMW_QOS_POLICY_LIVELINESS_SYSTEM_DEFAULT: 0>
>>> QoSLivelinessPolicy['automatic']
<QoSLivelinessPolicy.RMW_QOS_POLICY_LIVELINESS_AUTOMATIC: 1>
```

The keys can be gotten off the class via `__members__`. One drawback for a CLI tool is the values would be a bit noisy as it includes the long all capitals name.

```python3
>>> QoSLivelinessPolicy.__members__.keys()
odict_keys(['RMW_QOS_POLICY_LIVELINESS_SYSTEM_DEFAULT', 'system_default', 'RMW_QOS_POLICY_LIVELINESS_AUTOMATIC', 'automatic', 'RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_NODE', 'manual_by_node', 'RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC', 'manual_by_topic'])
```

But that could be cleaned up with a little filtering

```python3
>>> [k for k in QoSLivelinessPolicy.__members__.keys() if not k.startswith('RMW')]
['system_default', 'automatic', 'manual_by_node', 'manual_by_topic']
```

Reply:
I like this suggestion, thanks! I have pushed a new commit to use this pattern, what do you think?

Reply:
Would these still be useful for the CLI if the names were `SYSTEM_DEFAULT` instead of `system_default`? Since [PEP 8 wants all-caps](https://www.python.org/dev/peps/pep-0008/#constants), it seems likely a linter will complain about these being lower case now or in the future.

The enum documentation also [argues in favor of all-caps](https://docs.python.org/3/library/enum.html#enum-member-type).

Reply:
I think it would be a reasonable use case to have users call `PolicyEnum[arg.upper()]`, or perhaps provide another convenience function `QoSPolicyEnum.get_short_name(cls, name)` that uppercases the name

Reply:
>  or perhaps provide another convenience function QoSPolicyEnum.get_short_name(cls, name) that uppercases the name

Did you mean a convenience function that provides lower case names? I suspect the linter tests in this package will fail in the future if `system_default` etc is lowercase instead of uppercase, meaning this might need to be

```python3
class QoSLivelinessPolicy(IntEnum):
    """
    Enum for QoS Liveliness settings.
    This enum matches the one defined in rmw/types.h
    """

    RMW_QOS_POLICY_LIVELINESS_SYSTEM_DEFAULT = 0
    SYSTEM_DEFAULT = RMW_QOS_POLICY_LIVELINESS_SYSTEM_DEFAULT 
    RMW_QOS_POLICY_LIVELINESS_AUTOMATIC = 1
    AUTOMATIC = RMW_QOS_POLICY_LIVELINESS_AUTOMATIC 
    RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_NODE = 2
    MANUAL_BY_NODE = RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_NODE 
    RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC = 3
    MANUAL_BY_TOPIC = RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC 
```

Reply:
New commit should clarify what I was trying to say : yes use all uppercase names for enum, but provide case-insensitive getter function for users.